plugins/module_utils/utils.py

```diff
@@ -518,8 +518,7 @@ def _version_check_greater(self, greater, lesser, greater_or_equal=False):
 
     def _connect_api(self, url, token, ssl_verify, api_version):
         try:
-            nb = pynautobot.api(url, token=token, api_version=api_version)
-            nb.http_session.verify = ssl_verify
+            nb = pynautobot.api(url, token=token, api_version=api_version, verify=ssl_verify)
```

Comment:
We need to convert this entire method to catch the error better now that pynautobot checks for the version during instantiation.

I pretty much just copied what [we are doing in develop](https://github.com/nautobot/nautobot-ansible/blob/develop/plugins/module_utils/utils.py#L531-L542) for this:
```python
    def _connect_api(self, url, token, ssl_verify, api_version):
        try:
            nb = pynautobot.api(url, token=token, api_version=api_version, verify=ssl_verify)
            self.version = nb.version
            return nb
        except pynautobot.RequestError as e:
            self._handle_errors(msg=e.error)
        except ValueError as e:
            # pynautobot does version constraint on init, handle errors if versions doesn't match.
            self._handle_errors(msg=str(e))
        except Exception:
            self.module.fail_json(msg="Failed to establish connection to Nautobot API")
```